### ISSUE TITLE
Form elements always has value from setDefault method

### ISSUE THREAD

**Issue by pronigo** [phalcon/cphalcon#3235]
If you create a form element and use setDefault method, the element has always value you provided in setDefault method. After form validation with errors, all elements in form should have values from POST array, but have default values set by setDefault method.

My code:

```
// FORM
$quantity = new Numeric('quantity');
$quantity->addValidators(array(
    new PresenceOf(),
    new Between(array(
        'minimum' => 1,
        'maximum' => 100
    ))
));
$quantity->setDefault(1);
$this->add($quantity);

// CONTROLLER
$form = new DocumentForm();

if ($this->request->isPost() && $form->isValid($this->request->getPost())) {
    $document = new Document();
    $form->bind($this->request->getPost(), $document);
    $document->save();
    // redirect etc.
}

// in case of errors in forms, we show again form with errors
$this->view->setVar('form', $form);

// VIEW
<?=$form->renderDecorated('quantity')?>

I use renderDecorated method from Phalcon Doc.
```
So at the start, this field has value "1". I change it to for example "123" and submit the form. It shows me again the form with error and the value of this field is again "1" when it should be "123", does it right?

I'm quite sure it's a bug.

**Comment by sergeyklay** [comment]
Try this

```php
$form = new DocumentForm();
$document = new Document();

if ($this->request->isPost()) {
    // validate in model
    $document->assign($this->request->getPost());

    if ($document->save()) {
        //  flash, redirect, etc.
    } else {
        // flash, etc.
    }
} else {
    $form->setEntity($document);
}
```

**Comment by 7thcubic** [comment]
its a bug.
https://github.com/phalcon/cphalcon/blob/2.0.0/phalcon/forms/form.zep#L515


/**
	 * Renders a specific item in the form
	 *
	 * @param string name
	 * @param array attributes
	 * @return string
	 */
	public function render(string! name, var attributes = null) -> string
	{
		var element, method, entity, data, value;

		if !fetch element, this->_elements[name] {
			throw new Exception("Element with ID=" . name . " is not part of the form");
		}
		let entity = this->_entity;
		let data = this->_data;
		if typeof entity == "object" {
			let method = "get" . key;
			if method_exists(entity, method) {
				let attributes["value"] = entity->{method}();
			}
			else {
				let attributes["value"] = entity->{name};
			}
		}
		else if typeof data == "array" {
			if fetch value, data[name] {
				let attributes["value"] = value;
			}
		}
		return element->render(attributes);
	}

**Comment by pronigo** [comment]
Thank you, but unfortunately it doesn't work. I made some hot fix in renderDecorated() method, but it's not pretty solution...

````
public function renderDecorated($name)
{
    $return = '';
    $element = $this->get($name);

    if ($this->request->getPost($name)) {
        $element->setDefault($this->request->getPost($name));
    }

    ...
````

**Comment by 7thcubic** [comment]
i think the error occurs here

https://github.com/phalcon/cphalcon/blob/2.0.0/phalcon/forms/element.zep#L481

**Comment by pronigo** [comment]
It looks like you're right. Let's wait now for Phalcon comment.

**Comment by olivier-monaco** [comment]
See https://github.com/phalcon/cphalcon/issues/2559#issuecomment-57209190

**Comment by Green-Cat** [comment]
@andresgutierrez Fixed in #3330
